### PR TITLE
RUN-1346: remove db storage for "node filters"

### PR DESCRIPTION
**Pull request by gschueler**
**Is this a bugfix, or an enhancement? Please describe.**
Removes the db-backed storage for stored Node filters, replaces it with browser local storage

**Describe the solution you've implemented**

* Removes the db-backed storage for Node filters
* Replaces GSP UI for node filters with the Vue component
    * Commands page
    * Nodes page
    * Job edit > workflow edit > job reference
    * Job show page
* changes implementation for Node filters to use browser local storage

### CODE REVIEW

File: rundeckapp/grails-app/migrations/core/ConstraintsIndexesKeys.groovy [hunk 1]
```diff
@@ -543,6 +512,7 @@ databaseChangeLog = {
             not{
                 foreignKeyConstraintExists (foreignKeyTableName: "node_filter", foreignKeyName: "FKhuj3n3gy2kqoie8d8oucay4b3")
             }
+            tableExists (tableName:"node_filter")
         }
         addForeignKeyConstraint(baseColumnNames: "user_id", baseTableName: "node_filter", constraintName: "FKhuj3n3gy2kqoie8d8oucay4b3", deferrable: "false", initiallyDeferred: "false", referencedColumnNames: "id", referencedTableName: "rduser", validate: "true")
```
Comment (ronaveva) [inline rundeckapp/grails-app/migrations/core/ConstraintsIndexesKeys.groovy:517]: i think this changeset is no longer needed

File: rundeckapp/grails-app/migrations/core/H2AlterIdColumnsToAutoIncrementByDefault.groovy [hunk 2]
```diff
@@ -8,17 +8,17 @@ databaseChangeLog = {
                 sql.execute("alter table EXECUTION alter column id BIGINT GENERATED BY DEFAULT AS IDENTITY DEFAULT ON NULL;")
                 sql.execute("alter table JOB_FILE_RECORD alter column id BIGINT GENERATED BY DEFAULT AS IDENTITY DEFAULT ON NULL;")
                 sql.execute("alter table LOG_FILE_STORAGE_REQUEST alter column id BIGINT GENERATED BY DEFAULT AS IDENTITY DEFAULT ON NULL;")
-                sql.execute("alter table NODE_FILTER alter column id BIGINT GENERATED BY DEFAULT AS IDENTITY DEFAULT ON NULL;")
+//                sql.execute("alter table NODE_FILTER alter column id BIGINT GENERATED BY DEFAULT AS IDENTITY DEFAULT ON NULL;")
```
[the remark this answers is not in the document]
Comment (gschueler) [reply rundeckapp/grails-app/migrations/core/H2AlterIdColumnsToAutoIncrementByDefault.groovy]: I added the validCheckSum

File: rundeckapp/grails-app/migrations/core/ConstraintsIndexesKeys.groovy (hunk 1 above)
Comment (gschueler) [reply rundeckapp/grails-app/migrations/core/ConstraintsIndexesKeys.groovy]: removed

**Review by carlosrfranco — APPROVED** [review]
Tested in:
- Commands page
- Nodes page
- Job edit > workflow edit > job reference
- Job show page
- Starting app using an existent database to check migrations that were updated

**Pull request merged**